docs/api/specs/browserResidentDesktopAgents.md

```diff
@@ -0,0 +1,253 @@
+
+App identity is validated and an `appId` assigned by matching the application to an AppD record already known to the Desktop Agent. This is achieved by matching  `identityUrl` (supplied by the application via the `getAgent()` implementation) to the `details.url` field of the known App Directory records.
+
+As web applications may vary their URL during use, or serve multiple applications from the same origin (differentiated by path, search params and/or hash), care must be taken in matching URLs to appD records. Two URLs are sent to the Desktop Agent in the [`"WCP4ValidateAppIdentity"`](https://fdc3.finos.org/schemas/next/api/WCP4ValidateAppIdentity.schema.json) message:
+
+- `identityUrl`: the URL to match to the app directory record.
+- `actualUrl`: the current URL of the application, which MUST be captured automatically by the `getAgent()` implementation.
+
+Applications _may_ specify the `identityUrl` value as an argument to `getAgent()`. If not specified, the `getAgent()` implementation MUST use the current URL of the application. The Desktop Agent must validate that the origin of the `identityUrl` is the same as the origin of _both_ the `actualUrl` and the `WCPValidateAppIdentity` message sent over the `MessagePort`. The Desktop Agent MUST then match the URL to that of applications known to the Desktop Agent.
+
+The `actualUrl` field may be used for logging and debug purposes by the Desktop Agent and it differing from the `identityUrl` indicates that the application provided an override via `getAgent()`.
+
+Owing to the fact that the different parts of a URL (origin, path, search parameters, anchor) are used differently by web applications, matching of the `identityUrl` to known application URLs can be more complex than a simple string match. To allow application developers control the requirements of matching, Desktop Agents SHOULD consider a URL to match if all elements (origin, path, search parameters, anchor) present in the App Directory record's URL are present in the `identityUrl`. As multiple App Directory records may match a given identity URL, Desktop Agents SHOULD look for the best match that meets the requirements.
+
+For example, given an identity URL `url`, and an array of App Directory records `appDRecords`, a Desktop Agent MAY implement matching as follows:
```

Comment:
I guess this is labelled as MAY.  However, my takeaway from the discussion we had in the Web browsers meeting was that the point of sending the identity URL was that the app would send the URL that was listed in the AppD record.  But then we wouldn't need to do this kind of matching, would we?

Reply:
When we discussed this the consensus was that we wanted to try and support zero-argument use of getAgent and *any* changes or additions path, search params, or arguments would break an exact match - you and I reconfirmed that in some github comments a few days ago. 
However, if there's no possible fuzziness to the URL match then the identityUrl should probably become a required argument.

But as you say, this is down as a MAY, so it's allowed, but not even recommended (SHOULD) currently. I think it's something we should try out in practice, get feedback on and refine in a subsequent version (when or before we remove the @experimental label from this).

Personally, I still think a tighter link to a specific appD record would be better and necessary for future security/auth recommendations (likely based on async encryption). If we do that it would have to come with some sort of API call or challenge to provide a signed token, that could include identifying where the (appD) record is that contains the public key.  Its also likely to need more work on and adoption of the appD standard.